### PR TITLE
Include hand crafted resources in documentation indexes

### LINKED ISSUE #3054: Bug: Hand coded resources are omitted from CRD Documentation
**Describe the bug**

Hand written resources - like users for PostgreSQL databases - aren't included in our CRD documentation; they should be.

**To Reproduce**

Look at the [documentation for PostgreSQL](https://azure.github.io/azure-service-operator/reference/dbforpostgresql/) and observe that only API versions are listed. The `v1` version used for the hand-written resources does not appear.

**Expected behavior**

All resources should be documented, regardless of whether they are generated or hand-written.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Our index pages need to include any hand-crafted types that we add to ASO; without them, the documentation (which we _are_ generating) can't be easily found.

Closes #3054 

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/azure-arm.yaml [hunk 1]
```diff
@@ -938,6 +938,9 @@ objectModelConfiguration:
         $exportAs: Database
         $supportedFrom: v2.0.0-beta.1
   dbformysql:
+    v1: # Handcrafted
```
Comment (matthchr) [inline v2/azure-arm.yaml:941]: Do we have a way to detect if we've forgotten to add this?

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go [hunk 2]
```diff
@@ -98,19 +105,23 @@ func NewResourceVersionsReport(
...
 
 	err = result.loadFragments()
 	if err != nil {
 		return nil, errors.Wrapf(err, "Unable to load report fragments")
 	}
 
-	result.summarize(definitions)
+	err = result.summarize(definitions)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to summarize resources")
```
Comment (matthchr) [inline v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go:122]: super super minor: I think Go prefers errs not uppercase first letter. Who knows why...

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go [hunk 3]
```diff
@@ -521,10 +543,21 @@ func (report *ResourceVersionsReport) FindSampleLinks(group string) (map[string]
...
+		name:          name,
+		armType:       armType,
+		armVersion:    armVersion,
+		supportedFrom: report.supportedFrom(name),
+	}
+}
 
-	name := rsrc.Name()
+// generateApiLink returns a link to the API definition for the given resource
+func (report *ResourceVersionsReport) generateApiLink(name astmodel.TypeName) string {
```
Comment (matthchr) [inline v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go:560]: minor: `API` preferred I think

**Review by matthchr — COMMENTED** [review]
Overall LGTM. One question is if there's a way for us to detect that somebody has forgotten to include a handcrafted resource in the `azure-arm.yaml` file?

Doesn't necessarily have to be part of the code generator (in fact in some ways that'd be sorta weird). Maybe some way to write a taskfile target that verifies this?

The more I think about it the more I wonder if we should split the doc generation out into a different tool that does something based more on the filesystem (which would then work more easily for handcrafted resources as they also exist on the filesystem)... something to think about longer term possibly.

File: v2/azure-arm.yaml (hunk 1 above)
Comment (theunrepentantgeek) [reply v2/azure-arm.yaml:941]: I'm not sure at the moment. I'll give that some thought though.

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go (hunk 2 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go:122]: Fixed.

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go (hunk 3 above)
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go]: Fixed.

**Review by matthchr — APPROVED** [review]

**Pull request merged**